### PR TITLE
feat(domain/operation): add machine task query by status

### PR DESCRIPTION
**Pull request by gfouillet**
Introduce a new service method `GetMachineTaskIDsWithStatus` in `domain/operation/service/query.go` to retrieve task IDs filtered by machine name and status.

This is the minimal required implementation: despite the facade provides a complete task result, the only usage of this facade only fetches for TaskID. So this implementation doesn't return full-fledged TaskInfo, but only the required TaskID.

- Implemented `GetMachineTaskIDsWithStatus` in `service.go` and `query.go` to delegate state query operations to `State`. Validation of a machine name is performed via `coremachine.Name.Validate`.
- Added a state-level method `GetMachineTaskIDsWithStatus` in `state/task.go`, leveraging database queries to retrieve task IDs filtered by machine and status.
- Enhanced testing with multiple scenarios:
  - `query_test.go`: Unit tests for service behavior, validation errors, and state error propagation.
  - `task_test.go`: Database-level tests covering task ID retrieval, filtering by machine and status, and edge cases with no results.
- Updated mock generation for state and service components in `state_mock_test.go` and `package_mock_test.go`.

Additionally, integrated this capability into the API layer, specifically in `apiserver/facades/agent/machineactions/machineactions.go`. New functionality in `RunningActions` now uses this query to retrieve running actions for machines. Corresponding tests added in `machineactions_test.go` for API-level verification.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

 Can't QA yet, Unit Test should pass

## Links

**Jira card:** JUJU-8544

### CODE REVIEW

File: apiserver/facades/agent/machineactions/machineactions.go [hunk 1]
```diff
@@ -214,10 +220,51 @@ func (f *Facade) WatchActionNotifications(ctx context.Context, args params.Entit
 // RunningActions lists the actions running for the entities passed in.
 // If we end up needing more than ListRunning at some point we could follow/abstract
 // what's done in the client actions package.
+//
+// Note(gfouillet): this implementation is minimal to fulfill requirements for
```
Comment (hmlanigan) [inline apiserver/facades/agent/machineactions/machineactions.go:223]: 💯

File: apiserver/facades/agent/machineactions/machineactions.go [hunk 2]
```diff
@@ -45,6 +47,10 @@ type OperationService interface {
 	// NOTE: This watcher will emit events for tasks changing their statuses to
 	// PENDING only.
 	WatchMachineTaskNotifications(ctx context.Context, machineName coremachine.Name) (watcher.StringsWatcher, error)
+
+	// GetMachineTaskIDsWithStatus retrieves all task IDs for a machine with the specified status.
+	// Returns an error if the operation fails.
```
Comment (hmlanigan) [inline apiserver/facades/agent/machineactions/machineactions.go]: nit: please rephrase without the work `operation`. It's used in a different context already in this domain.

File: apiserver/facades/agent/machineactions/machineactions_test.go [hunk 3]
```diff
@@ -179,6 +175,75 @@ func (s *facadeSuite) TestActionsTaskNotPending(c *tc.C) {
...
+	c.Check(results.Actions[0].Actions[1].Action.Tag, tc.Equals, names.NewActionTag("47").String())
+}
+
+func (s *facadeSuite) TestRunningActionsReturnsActionIDsForMachineEmptyArgs(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	// Arrange
+	s.operationService.EXPECT().GetMachineTaskIDsWithStatus(
+		gomock.Any(), coremachine.Name(s.authTag.Id()), status.Running,
+	).Times(0) // not called
```
Comment (hmlanigan) [inline apiserver/facades/agent/machineactions/machineactions_test.go]: delete me: this isn't needed, the test will already fail if `GetMachineTaskIDsWithStatus` is called.

File: domain/operation/service/query.go [hunk 4]
```diff
@@ -20,3 +22,19 @@ func (s *Service) GetOperations(ctx context.Context, params operation.QueryArgs)
 func (s *Service) GetOperationByID(ctx context.Context, operationID string) (operation.OperationInfo, error) {
 	return operation.OperationInfo{}, errors.New("actions in Dqlite not supported")
 }
+
+// GetMachineTaskIDsWithStatus retrieves the task IDs for a specific machine name and status.
+func (s *Service) GetMachineTaskIDsWithStatus(ctx context.Context,
+	name coremachine.Name, running corestatus.Status) ([]string, error) {
```
Comment (hmlanigan) [inline domain/operation/service/query.go]: thought: you can validate the status as one for operations and tasks with `if running.KnownTaskStatus() {}`

**Review by hmlanigan — APPROVED** [review]

File: apiserver/facades/agent/machineactions/machineactions_test.go (hunk 3 above)
Comment (gfouillet) [reply apiserver/facades/agent/machineactions/machineactions_test.go]: That was a test to see if the error message would be better or not, and it got out of my mind.
Thanks for spotting it !

File: apiserver/facades/agent/machineactions/machineactions.go (hunk 2 above)
Comment (gfouillet) [reply apiserver/facades/agent/machineactions/machineactions.go]: indeed :smiling_face_with_tear:

File: domain/operation/service/query.go [hunk 5]
```diff
@@ -20,3 +23,23 @@ func (s *Service) GetOperations(ctx context.Context, params operation.QueryArgs)
...
```
Comment (nvinuesa) [inline domain/operation/service/query.go]: Why is this arg named `running`? It's just *any* status right?

File: domain/operation/state/task.go [hunk 6]
```diff
@@ -47,6 +47,52 @@ func (s *State) GetTask(ctx context.Context, taskID string) (operation.Task, *st
...
+func (s *State) GetMachineTaskIDsWithStatus(ctx context.Context, machineName string, statusFilter string) ([]string, error) {
+	db, err := s.DB(ctx)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type search struct {
+		Machine string `db:"machine"`
+		Status  string `db:"status"`
+	}
```
Comment (nvinuesa) [inline domain/operation/state/task.go:62]: Move me to types.go.

File: domain/operation/service/query.go (hunk 5 above)
Comment (gfouillet) [reply domain/operation/service/query.go]: intellij default name when i do "create method". I usually write thing top down. Thanks for spotting it will change :D

File: domain/operation/state/task.go (hunk 6 above)
Comment (gfouillet) [reply domain/operation/state/task.go:62]: This one is only inside the function, this is not desirable (very contextual so far)
